resize-detector/src/main/scala/lucuma/react/resizeDetector/hooks.scala

```diff
@@ -3,95 +3,61 @@
-    val height = r.height.toOption.map(_.toInt)
-    val width  = r.width.toOption.map(_.toInt)
-    val ref    =
-      Ref.fromJs(r.ref)
-  }
+    trait UseResizeObserverOptions extends js.Object
 
-  implicit val reusabilityUseResizeDetectorReturn: Reusability[UseResizeDetectorReturn] =
-    Reusability.by(x => (x.height, x.width, x.ref))
-}
-
-@js.native
-protected trait UseResizeDetectorProps extends Props {
-  var targetRef: facade.React.RefFn[html.Element]
-}
-
-object UseResizeDetectorProps {
-
-  def apply(
-    onResize:        js.UndefOr[(Int, Int) => Callback] = js.undefined,
-    handleHeight:    js.UndefOr[Boolean] = js.undefined,
-    handleWidth:     js.UndefOr[Boolean] = js.undefined,
-    skipOnMount:     js.UndefOr[Boolean] = js.undefined,
-    refreshMode:     js.UndefOr[RefreshMode] = js.undefined,
-    refreshRate:     js.UndefOr[Int] = js.undefined,
-    refreshOptions:  js.UndefOr[RefreshOptions] = js.undefined,
-    observerOptions: js.UndefOr[ObserverOptions] = js.undefined
-  ): UseResizeDetectorProps = {
-    val p = (new js.Object).asInstanceOf[UseResizeDetectorProps]
-    onResize.foreach(v =>
-      p.onResize = { (x, y) =>
-        if (x.isDefined && y.isDefined) v(x.get.toInt, y.get.toInt).runNow()
-      }: js.Function2[js.UndefOr[Double], js.UndefOr[Double], Unit]
-    )
-    handleHeight.foreach(v => p.handleHeight = v)
-    handleWidth.foreach(v => p.handleWidth = v)
-    skipOnMount.foreach(v => p.skipOnMount = v)
-    refreshMode.toJs.foreach(v => p.refreshMode = v)
-    refreshRate.foreach(v => p.refreshRate = v)
-    refreshOptions.foreach(v => p.refreshOptions = v)
-    observerOptions.foreach(v => p.observerOptions = v)
-    p
-  }
-}
-
-object HooksApiExt {
-
-  object mod {
     @js.native
-    @JSImport("react-resize-detector", "useResizeDetector")
-    def baseUseResizeDetector: js.Function1[
-      js.UndefOr[UseResizeDetectorProps],
-      UseResizeDetectorReturnJS
+    @JSImport("@react-hook/resize-observer", JSImport.Default)
+    val useResizeObserver: js.Function2[
+      facade.React.RefHandle[TopNode | Null],
+      UseResizeObserverCallback,
+      ResizeObserver
     ] = js.native
-
-    inline def useResizeDetector(
-      props: js.UndefOr[UseResizeDetectorProps] = js.undefined
-    ): UseResizeDetectorReturn =
-      UseResizeDetectorReturn.fromJS(baseUseResizeDetector(props))
   }
 
+  val jsHook =
+    CustomHook.unchecked(
+      (props: (facade.React.RefHandle[TopNode | Null], raw.UseResizeObserverCallback)) =>
+        raw.useResizeObserver(props._1, props._2)
+    )
+
   val hook =
     CustomHook[UseResizeDetectorProps]
-      .buildReturning { pos =>
-        mod.useResizeDetector(pos)
-      }
+      .useRefToVdom[HTMLElement]
+      .useStateBy((_, ref) => UseResizeDetectorReturn(none, none, ref))
+      .customBy: (props, ref, size) =>
+        jsHook(
+          (
+            ref.raw,
+            (entry, _) =>
+              val height = entry.contentRect.height.toInt
```

Comment:
Is it always defined? Maybe we can just pass the Int instead of `Option[Int]` and change it upstream?
Or does it have a null state sometimes?

Reply:
With JS it's hard to know. I think it will always be defined. This is from the DOM's `DOMRectReadOnly`, specified in https://drafts.fxtf.org/geometry-1/#domrectreadonly as:
```
readonly attribute unrestricted double width;
readonly attribute unrestricted double height;
```